src/main/java/org/cbioportal/service/impl/ClinicalDataServiceImpl.java

```diff
@@ -227,17 +234,44 @@ public List<ClinicalData> getPatientClinicalDataDetailedToSample(List<String> st
     }
 
     @Override
-    public List<ClinicalData> fetchSampleClinicalTable(List<String> studyIds, List<String> sampleIds, Integer pageSize,
-                                                       Integer pageNumber, String searchTerm, String sortBy,
-                                                       String direction) {
-        return clinicalDataRepository.fetchSampleClinicalTable(studyIds, sampleIds, pageSize, pageNumber, searchTerm,
-                                                                            sortBy, direction);
-    }
+    public ImmutablePair<SampleClinicalDataCollection, Integer> fetchSampleClinicalTable(List<String> studyIds, List<String> sampleIds, Integer pageSize, Integer pageNumber, String searchTerm, String sortBy, String direction) {
 
-    @Override
-    public Integer fetchSampleClinicalTableCount(List<String> studyIds, List<String> sampleIds,
-                                                 String searchTerm, String sortBy, String direction) {
-        return clinicalDataRepository.fetchSampleClinicalTableCount(studyIds, sampleIds, searchTerm, 
-            sortBy, direction);
+        SampleClinicalDataCollection sampleClinicalDataCollection = new SampleClinicalDataCollection();
+        if (studyIds == null || studyIds.isEmpty() || sampleIds == null || sampleIds.isEmpty()) {
+            return new ImmutablePair<>(sampleClinicalDataCollection, 0);
+        }
+        
+        // Request un-paginated data.
+        List<Integer> allSampleInternalIds = clinicalDataRepository.getVisibleSampleInternalIdsForClinicalTable(
+            studyIds, sampleIds,
+            null, null,
+            searchTerm, sortBy, direction
+        );
+        Integer offset = paginationCalculator.offset(pageSize, pageNumber);
```

Comment:
See above.
Then this can just be a static method. :-)

Reply:
@JREastonMarks  I agree. However, I merely changed and renamed the exisiting OffsetCalculator class. I would propose to change this in a followup pull request.